autogen/agentchat/contrib/graph_rag/graph_rag_agent.py

```diff
@@ -0,0 +1,44 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+from autogen.agentchat import ConversableAgent
+
+from .graph_store import GraphStore
+
+
+class GraphRagAgent(ConversableAgent, ABC):
+    """
+    A graph rag agent is a conversable agent which could query graph database for answers.
+
+    An implementing agent class would
+    1. create a graph in the underlying database with input documents
+    2. use the retrieve() method to retrieve information.
+    3. use the retrieved information to generate and send back messages.
+    """
+
+    @abstractmethod
+    def _init_db(self, input_doc: List | None = None) -> GraphStore:
```

Comment:
Could you add more details to the `input_doc`, such as `List[str]` or else?

Reply:
Updated